beacon_node/operation_pool/src/lib.rs

```diff
@@ -284,6 +283,16 @@ impl<E: EthSpec> OperationPool<E> {
         let mut num_prev_valid = 0_i64;
         let mut num_curr_valid = 0_i64;
 
+        // TODO(electra): Work out how to do this more elegantly. This is a bit of a hack.
+        let mut all_attestations = self.attestations.write();
+
+        if fork_name >= ForkName::Electra {
+            all_attestations.aggregate_across_committees(prev_epoch_key);
+            all_attestations.aggregate_across_committees(curr_epoch_key);
```

Comment:
Why do you need to persist the result of aggregating the attestations across committees in the pool? You can just retrieve the best attestation on each committee, merge them and return the result for block production

Reply:
yes the TODO above mentions we should find a way to do this more elegantly. I agree that persisting the final aggregation in the pool is unnecessary. I'll open up a separate issue for this